components/layout/block.rs

```diff
@@ -324,22 +324,20 @@ impl CandidateBSizeIterator {
             (LengthOrPercentageOrAuto::Percentage(percent), Some(block_container_block_size)) => {
                 MaybeAuto::Specified(block_container_block_size.scale_by(percent))
             }
-            (LengthOrPercentageOrAuto::Calc(calc), Some(block_container_block_size)) => {
-                MaybeAuto::Specified(calc.length() + block_container_block_size.scale_by(calc.percentage()))
+            (LengthOrPercentageOrAuto::Calc(calc), block_container_block_size) => {
```

Comment:
No need to bind a new name here; just write `(LengthOrPercentageOrAuto::Calc(calc), _)` and use the variable name in the match expression.

Same for the below.